File: javascript/space-age/space-age.js
var SpaceAge = function(value) {
  this.seconds = value;
};

SpaceAge.prototype.seconds = 0;

SpaceAge.prototype.getAge = function(factor) {
  return Math.round(this.seconds / factor * 100) / 100;
};

SpaceAge.prototype.onEarth   = function() { return this.getAge(31557600); };
SpaceAge.prototype.onMercury = function() { return this.getAge(7600543.81992); };
SpaceAge.prototype.onVenus   = function() { return this.getAge(19414149.052176); };
SpaceAge.prototype.onMars    = function() { return this.getAge(59354032.69008); };
SpaceAge.prototype.onJupiter = function() { return this.getAge(374355659.124); };
SpaceAge.prototype.onSaturn  = function() { return this.getAge(929292362.8848); };
SpaceAge.prototype.onUranus  = function() { return this.getAge(2651370019.3296); };
SpaceAge.prototype.onNeptune = function() { return this.getAge(5200418560.032); };

module.exports = SpaceAge;

File: python/space-age/space_age.py
class SpaceAge(object):
    def __init__(self, seconds):
        self.seconds = seconds


    def converter(period):
        def inner(self):
            return round(self.seconds/period, 2)

        return inner


    on_earth   = converter(31558149.76)
    on_jupiter = converter(374335776.0)
    on_mars    = converter(59354294.4)
    on_mercury = converter(7600530.24)
    on_neptune = converter(5200418592.0)
    on_saturn  = converter(929596608.0)
    on_uranus  = converter(2661041808.0)
    on_venus   = converter(19413907.2)
